src/clientrpc.cpp

```diff
@@ -0,0 +1,201 @@
+ * this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA
+ *
+ \******************************************************************************/
+
+#include "clientrpc.h"
+
+CClientRpc::CClientRpc ( QObject* parent, CClient* pClient, CRpcServer* pRpcServer ) : QObject ( parent )
+{
+    connect ( pClient, &CClient::ChatTextReceived, [=] ( QString strChatText ) {
+        pRpcServer->BroadcastNotification ( "jamulusclient/chatTextReceived",
+                                            QJsonObject{
+                                                { "chatText", strChatText },
+                                            } );
+    } );
+
+    connect ( pClient, &CClient::ClientIDReceived, [=] ( int iChanID ) {
+        pRpcServer->BroadcastNotification ( "jamulusclient/connected",
+                                            QJsonObject{
+                                                { "id", iChanID },
+                                            } );
+    } );
+
+    connect ( pClient, &CClient::ConClientListMesReceived, [=] ( CVector<CChannelInfo> vecChanInfo ) {
+        QJsonArray arrChanInfo;
+        for ( const auto& chanInfo : vecChanInfo )
+        {
+            QJsonObject objChanInfo{
+                { "id", chanInfo.iChanID },
+                { "name", chanInfo.strName },
+                { "skillLevel", SerializeSkillLevel ( chanInfo.eSkillLevel ) },
+                { "countryId", chanInfo.eCountry },
+                { "city", chanInfo.strCity },
+                { "instrumentId", chanInfo.iInstrument },
+            };
+            arrChanInfo.append ( objChanInfo );
+        }
+        pRpcServer->BroadcastNotification ( "jamulusclient/clientListReceived",
+                                            QJsonObject{
+                                                { "clients", arrChanInfo },
+                                            } );
+        arrStoredChanInfo = arrChanInfo;
+    } );
+
+    connect ( pClient, &CClient::CLChannelLevelListReceived, [=] ( CHostAddress /* unused */, CVector<uint16_t> vecLevelList ) {
+        QJsonArray arrLevelList;
+        for ( const auto& level : vecLevelList )
+        {
+            arrLevelList.append ( level );
+        }
+        pRpcServer->BroadcastNotification ( "jamulusclient/channelLevelListReceived",
+                                            QJsonObject{
+                                                { "channelLevelList", arrLevelList },
+                                            } );
+    } );
+
+    connect ( pClient, &CClient::Disconnected, [=]() { pRpcServer->BroadcastNotification ( "jamulusclient/disconnected", QJsonObject{} ); } );
+
+    pRpcServer->HandleMethod ( "jamulus/getMode", [=] ( const QJsonObject& params, QJsonObject& response ) {
+        QJsonObject result{ { "mode", "client" } };
+        response["result"] = result;
+        Q_UNUSED ( params );
```

Comment:
I'm in two minds on this - part of me wants not to provide a named argument to the lambda only for it to be unused...

Reply:
Having worked some more with this code, the number of `Q_UNUSED ( params )` instances starts to grow. However, the solution I thought above is not as straight-forward. Basically, we would have to duplicate the whole logic (mappings, etc.) to cover `params`-less methods as well. Not sure if it's worth it?